packages/frontend/src/features/catalog/components/CatalogPanel.tsx

```diff
@@ -41,53 +38,71 @@ export const CatalogPanel: FC<React.PropsWithChildren<Props>> = ({
         search: debouncedSearch,
     });
 
-    const [selectedTable, setSelectedTable] = useState<string>();
+    // TODO: add field
+    const [selection, setSelection] = useState<{
+        group: string;
+        table: string;
+    }>();
+
     const {
         mutate: getMetadata,
         data: metadata,
         reset: closeMetadata,
     } = useCatalogMetadata(projectUuid);
 
-    const [catalogGroupMap, ungroupedCatalogItems] = useMemo(() => {
-        if (catalogResults) {
-            const results = catalogResults.reduce<
-                [Record<string, CatalogItem[]>, CatalogItem[]]
-            >(
-                (acc, item) => {
-                    if ('groupLabel' in item && item.groupLabel) {
-                        return [
-                            {
-                                ...acc[0],
-                                [item.groupLabel]: acc[0][item.groupLabel]
-                                    ? [...acc[0][item.groupLabel], item]
-                                    : [item],
-                            },
-                            acc[1],
-                        ];
-                    }
-                    return [acc[0], [...acc[1], item]];
-                },
-                [{}, []],
-            );
-            return [
-                results[0],
-                results[1].sort((a, b) => a.name.localeCompare(b.name)),
-            ];
-        }
-        return [{}, []];
-    }, [catalogResults]);
-
     const handleSearchChange = useCallback(
         (searchString: string) => {
             setSearch(searchString);
         },
         [setSearch],
     );
 
+    // TODO: should this transform be in the backend?
```

Comment:
yeah, I'm not sure about this. Is it better to return the data formatted from the backend ? This means this endpoint will be less reusable and I dont' think this will make it more efficient though (we're just moving the complexity to the backend and making search slower...) What do you think ?

Reply:
I think it might depend on the perf we end up seeing with this. This tree function is expensive, but I put some effort into writing it as a single pass instead of multiple, so it might be fine. If it seems to hand the UI thread, we should move it to the BE. Otherwise, I agree that leaving it here makes things more flexible.